backpack/__init__.py

```diff
@@ -43,6 +44,17 @@ def __init__(self, *exts: BackpropExtension, debug=False):
         Attributes:
             args: [BackpropExtension]
                 The extensions to activate for the backward pass.
+            extension_hook: Callable, optional (default: None)
+                Function called on each module after all BackPACK extensions have run. 
+                Takes a ``torch.nn.Module`` and returns ``None``. 
+                
+                Can be used to reduce memory overhead if the goal is to compute
+                transformations of BackPACK quantities. Information can be compacted
+                during a backward pass and obsolete tensors be freed manually (``del``).
+
+                Note: 
+                    If the callable iterates over the module parameters, it may iterate
+                    multiple times over some since the hook acts on the modular level.
```

Comment:
```
If the callable iterates over the module parameters, it may iterate
multiple times over some since the hook acts on the modular level.
```
Not sure I follow. If all backpack hooks (including this one) act on the module level, it will be called once and iterate over the module parameter once? Did you mean `model` or am I missing something?

Reply:
Depending on what is happening here, ``test/extensions/implementation/hooks.py`` might need some tweaks

Reply:
I added [tests to demonstrate the subtleties of extension hooks](https://github.com/f-dangel/backpack/blob/0d5f97e5ca93baff371326c28a30ae5df22b287d/test/extensions/test_hooks.py). The origin is that an extension hook will also be called from the backward hook attached to a `nn.Sequential`.